velox/functions/sparksql/DecimalVectorFunctions.cpp

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "velox/functions/sparksql/DecimalVectorFunctions.h"
+
+#include "velox/expression/DecodedArgs.h"
+
+namespace facebook::velox::functions::sparksql {
+namespace {
+
+// Return the unscaled bigint value of a decimal, assuming it
+// fits in a bigint. Only short decimal input is accepted.
+class UnscaledValueFunction final : public exec::VectorFunction {
+  void apply(
+      const SelectivityVector& rows,
+      std::vector<VectorPtr>& args,
+      const TypePtr& outputType,
+      exec::EvalCtx& context,
+      VectorPtr& result) const final {
+    VELOX_CHECK_EQ(args.size(), 1);
+    VELOX_CHECK(
+        args[0]->type()->isShortDecimal(), "ShortDecimal type is required.");
+    exec::DecodedArgs decodedArgs(rows, args, context);
+    auto decimalVector = decodedArgs.at(0);
+    context.ensureWritable(rows, BIGINT(), result);
+    result->clearNulls(rows);
+    auto flatResult =
+        result->asUnchecked<FlatVector<int64_t>>()->mutableRawValues();
+    rows.applyToSelected([&](auto row) {
+      flatResult[row] = decimalVector->valueAt<int64_t>(row);
+    });
+  }
+};
+} // namespace
+
+std::vector<std::shared_ptr<exec::FunctionSignature>>
+unscaledValueSignatures() {
+  return {exec::FunctionSignatureBuilder()
+              .integerVariable("a_precision", "min(18, a_precision + 1)")
```

Comment:
How is "min(18, a_precision + 1)" used? It is needed? I assume it can be removed, no?

Reply:
I want to be able to limit a_precision to be less than or equal to 18.

Reply:
I'm not sure if this works as you expect? Can you provide an example of how this works?

Reply:
> I'm not sure if this works as you expect? Can you provide an example of how this works?

I tested this method and found it to be infeasible. I have removed the constraint parameter, and currently, the short decimal validation is still being done within the function itself. However, a better approach would be to have function signatures that support restrictions on decimal types.